frontend/packages/pipelines-plugin/src/components/shared/workspaces/VolumeClaimTemplateLink.tsx

```diff
@@ -0,0 +1,41 @@
+import * as React from 'react';
+import { useK8sWatchResource } from '@console/internal/components/utils/k8s-watch-hook';
+import { K8sResourceKind } from '@console/internal/module/k8s';
+import { PersistentVolumeClaimModel } from '@console/internal/models';
+import { ResourceLink } from '@console/internal/components/utils';
+
+export interface VolumeClaimTemplateLinkProps {
+  labels: { [key: string]: string };
+  workspaceName: string;
+  namespace: string;
+}
+
+const VolumeClaimTemplateLink: React.FC<VolumeClaimTemplateLinkProps> = ({
+  labels,
+  workspaceName,
+  namespace,
+}) => {
+  const [pvcResource, loaded, loadError] = useK8sWatchResource<K8sResourceKind>({
+    kind: PersistentVolumeClaimModel.kind,
+    isList: false,
+    selector: { matchLabels: labels },
+  });
+
+  if (loaded && !loadError && pvcResource?.metadata) {
```

Comment:
Nit: for readability, you typically want to return `null`s early so that the proper render is always at the bottom. It obv doesn't impact the flow, but it helps to be consistent in a codebase this large.

Reply:
thanks for the suggestion. Updated :)